client/systems.go

```diff
@@ -186,6 +186,20 @@ type SystemDetails struct {
 	Volumes map[string]*gadget.Volume `json:"volumes,omitempty"`
 
 	StorageEncryption *StorageEncryption `json:"storage-encryption,omitempty"`
+
+	// AvailableOptional contains the optional snaps and components that are
+	// available in this system.
+	AvailableOptional AvailableForInstall `json:"available-optional"`
+}
+
+// AvailableForInstall contains information about snaps and components that are
+// optional in the system's model, but are available for installation.
+type AvailableForInstall struct {
+	// Snaps contains the names of optional snaps that are available for installation.
+	Snaps []string `json:"snaps,omitempty"`
+	// Components contains a mapping of snap names to lists of the names of
+	// optional components that are available for installation.
+	Components map[string][]string `json:"components,omitempty"`
```

Comment:
in this world we could have a  `All bool json"all,omitempty"` so that we can do optional-install: {"all": true} which wouble be equivalent to omitting but is a bit more explicit to indicate that all available optional snaps/components should be installed?

Reply:
I've added this in 8a433a11858b2fc621279f379d8fc67e79ebd27d.